src/main/java/eu/adlogix/appnexus/oas/client/domain/Site.java

```diff
@@ -0,0 +1,40 @@
+package eu.adlogix.appnexus.oas.client.domain;
+
+import java.util.Map;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+public final class Site {
+
+	public interface Builder {
+		Site build(final String siteId);
+	}
+
+	@AllArgsConstructor
+	public static class SiteIdMapBackedBuilder implements Builder {
+		final Map<String, Site> siteMap;
```

Comment:
Without looking deeper, this seems very odd. Is it really Site's business to keep a temporary map?